source/code/include/scxcorelib/scxsingleton.h

```diff
@@ -58,21 +58,39 @@ namespace SCXCoreLib
             If the instance is not yet created, it first creates it.
 
         */
-        static T& Instance();
+        static T& Instance()
+        {
+            if (0 == s_lockHandle.GetData())
+            {
+                throw SCXInternalErrorException(L"Tried to get a singleton instance before static initialization was completed.", SCXSRCLOCATION);
+            }
+
+            SCXThreadLock lock(*s_lockHandle, true);
+
+            if (0 == s_instance.GetData())
+            {
+                s_instance = new T();
+            }
+
+            return *s_instance;
```

Comment:
I think this change here will partially override changes introduced by the commit: c01f6439e77435adb51ae230ca4add197c260122 .  

(I admit I do not see any concurrency issues - the definitions seem identical to me )